spec/rubocop/cop/layout/heredoc_indentation_spec.rb

```diff
@@ -268,6 +268,104 @@ def baz
         RUBY
       end
     end
+
+    context 'when `AllCops/ActiveSupportExtensionsEnabled: true`' do
+      let(:config) do
+        RuboCop::Config.new('AllCops' => { 'ActiveSupportExtensionsEnabled' => true })
+      end
+
+      it 'registers an offense for `squish` applied to heredoc' do
+        expect_offense(<<~RUBY)
+                    def foo
+                      <<-#{quote}RUBY2#{quote}.squish
+                      something
+          ^^^^^^^^^^^^^^^^^^^^^ Use 2 spaces for indentation in a heredoc by using `<<~` instead of `<<-`.
+                      RUBY2
+                    end
+        RUBY
+
+        expect_correction(<<-RUBY)
+          def foo
+            <<~#{quote}RUBY2#{quote}.squish
+              something
+            RUBY2
+          end
+        RUBY
+      end
+
+      it 'registers an offense for `squish` applied to heredoc when indentation is already good' do
+        expect_offense(<<~RUBY)
+                    def foo
+                      <<-#{quote}RUBY2#{quote}.squish
+                        something
+          ^^^^^^^^^^^^^^^^^^^^^^^ Use 2 spaces for indentation in a heredoc by using `<<~` instead of `<<-`.
+                      RUBY2
+                    end
+        RUBY
```

Comment:
In some regard I think this behavior falls out of the Layout jurisdiction and slips into the Style department, but changing `<<-` into `<<~` was already present before in this cop and I was just expanding upon it here.

My two cents are that there should be a separate Style cop for preferring `<<~` over `<<-` when possible (personally, in production projects I've worked on `<<-` was used only when devs didn't know about `<<~` already, and `<<-` treatment of indentation was always a nuisance).

Having such a cop would enable for example to add an "aggressive" mode to convert all instances of `<<-` into `<<~`, which is something not yet possible with RuboCop.